app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -60,17 +68,60 @@ export class GoogleCloudBuildStageForm extends React.Component<
     }));
   };
 
+  private getCloudBuildTriggers = (): Array<Option<string>> => {
+    return this.props.gcbTriggers.map(trigger => ({
+      label: trigger.name,
+      value: trigger.id,
+    }));
+  };
+
+  private getTriggerTypes = (): Array<Option<string>> => {
+    return [
+      { value: 'branchName', label: 'Branch name' },
+      { value: 'tagName', label: 'Tag Name' },
+      { value: 'commitSha', label: 'Commit SHA' },
+    ];
+  };
+
   private getSourceOptions = (): Array<Option<string>> => {
     return [
       { value: buildDefinitionSources.TEXT, label: 'Text' },
       { value: buildDefinitionSources.ARTIFACT, label: 'Artifact' },
+      { value: buildDefinitionSources.TRIGGER, label: 'Trigger' },
     ];
   };
 
   private onSourceChange = (e: any): void => {
     this.props.formik.setFieldValue('buildDefinitionSource', e.target.value);
   };
 
+  private onTriggerTypeValueChange = (e: any): void => {
+    const stage = this.props.formik.values;
+    if (stage.triggerType) {
```

Comment:
If a user inputs a value before selecting a type, will this just fail to work without any feedback? Can we default to the first type and add `clearable=false` to the type `Select` to avoid this potentially confusing behavior?

Reply:
If the user inputs a value without selecting a trigger, there should be an error displayed in the form. I implemented this here: https://github.com/spinnaker/deck/pull/7632/files#diff-c24618fef65c05c42e4c3a58f57f3b10R9
However, your suggestion makes sense. Will implement it.